### PR TITLE
[INTERNAL] Improve logging 

### PR DESCRIPTION
**Pull request by romaniam**
* Replace thrown errors with karma error logging. 
* Provide solution in case of error.

### CODE REVIEW

File: lib/framework.js [hunk 1]
```diff
@@ -187,11 +187,13 @@ class Framework {
 			});
 
 			if (!configs || configs.length === 0) {
-				throw new Error("Could not parse ui5.yaml");
+				karma.log("error", ["Could not parse ui5.yaml. Please make sure that the ui5.yaml has a valid format. For reference check: https://github.com/SAP/ui5-project/blob/master/docs/Configuration.md#configuration"]);
```
Comment (matz3) [inline lib/framework.js]: `karma` isn't available as a variable here.
We need to inject the `logger` and create a new logger for our framework.
`const log = logger.create('framework.ui5');`
You can take a look here how other plugins are doing so:
https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/ec01a4cacccd1eff9bc1dfcf322a98a825720de0/src/reporter.js#L19-L31

Comment (romaniam) [reply lib/framework.js]: Thanks. Is fixed in 742990e

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+module.exports = {
```
Comment (matz3) [inline .eslintrc.js]: I would prefer to use one tab for indentation

File: test/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (matz3) [inline test/.eslintrc]: This is only required within the integration tests, right?
So I would prefer to move this into `test/integration`.

Comment (romaniam) [reply test/.eslintrc]: done

File: package.json [hunk 4]
```diff
@@ -27,6 +27,8 @@
     "webpack.config.js"
   ],
   "scripts": {
+    "lint": "eslint ./",
+    "lint-fix": "eslint ./ --fix",
     "test": "npm run utest-coverage && npm run itest",
```
Comment (matz3) [inline package.json]: Linting should be part of the `npm test` script so that it is executed by Travis CI.

File: lib/client/.eslintrc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+	"extends": [
```
Comment (matz3) [inline lib/client/.eslintrc]: When not specifying `root: true`, the extends is not necessary as eslint reads up the folders until it finds a root eslintrc.
See: https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy

**Pull request merged**